libraries/AP_Compass/AP_Compass_HMC5843.cpp

```diff
@@ -282,12 +303,11 @@ void AP_Compass_HMC5843::read()
     publish_filtered_field(field, _compass_instance);
 }
 
-
-bool AP_Compass_HMC5843::_exit_test_mode()
+bool AP_Compass_HMC5843::_setup_mode()
```

Comment:
nops... it sets up the **mode** register so the sensor is put in the normal sampling mode rather than the calibration mode.